logbook-json/src/main/java/org/zalando/logbook/json/FastJsonHttpLogFormatter.java

```diff
@@ -12,24 +15,23 @@
 import org.zalando.logbook.HttpResponse;
 import org.zalando.logbook.Precorrelation;
 
-import java.io.IOException;
-import java.io.StringWriter;
-import java.util.List;
-import java.util.Map;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.ObjectMapper;
 
-import static org.apiguardian.api.API.Status.EXPERIMENTAL;
-import static org.apiguardian.api.API.Status.STABLE;
-import static org.zalando.logbook.Origin.LOCAL;
+import lombok.RequiredArgsConstructor;
 
 /**
  * A custom {@link HttpLogFormatter} that produces JSON objects.
  */
 @API(status = STABLE)
-@AllArgsConstructor
-public final class FastJsonHttpLogFormatter implements HttpLogFormatter {
+@RequiredArgsConstructor
+public final class FastJsonHttpLogFormatter implements HttpLogFormatter, JsonFieldWriter {
 
     private final JsonFactory factory;
 
+    private JsonFieldWriter delegate = this; // default to this implementation
```

Comment:
By implementing this interface directly you're exposing that implementation detail to the outside world because `FastJsonHttpLogFormatter` is public. I'd hide that with a static inner class.